iam_builder/iam_builder.py

```diff
@@ -3,70 +3,76 @@
 from iam_builder.templates import (
     iam_base_template,
     iam_lookup,
+    get_athena_read_access,
     get_pass_role_to_glue_policy,
     get_read_only_policy,
     get_write_only_policy,
     get_read_write_policy,
     get_s3_list_bucket_policy,
     get_secrets,
-    athena_dump_bucket,
 )
 
 
-def build_iam_policy(config):
+def build_iam_policy(config: dict) -> dict:
     """
     Takes a configuration for an IAM policy and returns the policy as a dict.
     """
-    iam = copy.deepcopy(iam_base_template)
+    iam: dict = copy.deepcopy(iam_base_template)
 
     list_buckets = []
+
     # Define if has athena permission
     if "athena" in config:
-        iam["Statement"].extend(iam_lookup["athena_read_access"])
+        dump_bucket = config["athena"].get("dump_bucket", ["mojap-athena-query-dump"])
+        if not isinstance(dump_bucket, list):
+            dump_bucket = [dump_bucket]
+        dump_bucket = [bucket.replace("_", "-") for bucket in dump_bucket]
+
+        iam["Statement"].extend(get_athena_read_access(dump_bucket))
 
         if config["athena"]["write"]:
             iam["Statement"].extend(iam_lookup["athena_write_access"])
 
         # Needed for s3tools package
-        list_buckets.append(athena_dump_bucket)
+        list_buckets.extend(dump_bucket)
 
     # Test to run glue jobs
     if "glue_job" in config and config["glue_job"]:
         iam["Statement"].extend(iam_lookup["glue_job"])
         # Add ability to pass itself to glue job
-        pass_role = get_pass_role_to_glue_policy(config["iam_role_name"])
+        pass_role: dict = get_pass_role_to_glue_policy(config["iam_role_name"])
```

Comment:
Not a right or wrong way to do this but my personal preference / thought on this is that if the function itself has type hints (for the return) then I don't think it's necessary to add them here. But like I said that's more a personal preference than anything else.

Reply:
Makes sense - I'll go with that approach and remove these unnecessary ones